replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/SyncHelper.java

```diff
@@ -139,7 +139,24 @@ public SyncHelper(
   }
 
   public SyncResponse sync() {
-    for (Result metacardResult : getMetacardChangeSet()) {
+    // Failed items and new items are replicated separately due to a bug in DDF that was only fixed
+    // recently
+    List<Filter> failedItemFilters = createFailedMetacardFilters();
+    if (!failedItemFilters.isEmpty()) {
+      replicateSubset(builder.anyOf(failedItemFilters));
+    }
+
+    // if the connection was lost while replicating failed items don't continue replicating
+    if (!status.getStatus().equals(Status.CONNECTION_LOST)) {
```

Comment:
❓ Should we skip this if replication was cancelled? Should it actually be `!= Status.SUCCESS`? The logic would be easier to read if the status would start at SUCCESS and then downgrade from that where it fails or is checked to be cancelled. Then finally aborted here if it is no longer success.

✏️ BTW, if Status is an enum then you should simply use `==` or `!=` and not `equals()`

Reply:
If we skip it if it is cancelled that would save us a query, I like that. 👍 The expected status would actually be PULL_IN_PROGRESS or PUSH_IN_PROGRESS and I think we use that to fetch the status for the UI so I didn't want to mess with that. Also will change to use `!=`